### PR TITLE
Move setup dependencies to pyproject.yaml

### PR DESCRIPTION
**Pull request by raddessi**
From https://github.com/smarie/python-pytest-cases/issues/65

Ahh now that I look at the diff my editor auto blacked the setup.py. If you want I can revert that.

### CODE REVIEW

**Comment by smarie** [comment]
> Ahh now that I look at the diff my editor auto blacked the setup.py. If you want I can revert that.

yes please do, so that the PR is "pure". Note that you should also leave the `SETUP_REQUIRES` untouched, because it is supposed to contain that info too (even if it is not used). So you can safely revert entirely the `setup.py`, and just leave the `.toml` file.

Once done I'll be glad to merge. Thanks again !

**Comment by raddessi** [comment]
Ah OK, I wasn't sure what the procedure was for setup.py. Great will do

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[build-system]
+requires = [
+    "setuptools",
+    "setuptools_scm",
+    "six",
+]
+build-backend = "setuptools.build_meta"
```
Comment (smarie) [inline pyproject.toml]: Not useful as per https://github.com/pypa/pip/issues/6453

```suggestion
```

**Comment by smarie** [comment]
thanks !

**Pull request merged**